src/utils/files/glob-source.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const fs = require('fs')
+const Path = require('path')
+const isString = require('lodash/isString')
+const pull = require('pull-stream')
+const glob = require('glob')
+const cat = require('pull-cat')
+const defer = require('pull-defer')
+const pushable = require('pull-pushable')
+const map = require('async/map')
+const parallel = require('async/parallel')
+const errCode = require('err-code')
+
+/**
+* Create a pull stream source that can be piped to ipfs.addPullStream for the
+* provided file paths.
+*
+* @param {String} ...paths File system path(s) to glob from
+* @param {Object} [options] Optional options
+* @param {Boolean} [options.recursive] Recursively glob all paths in directories
+* @param {Boolean} [options.hidden] Include .dot files in matched paths
+* @param {Array<String>} [options.ignore] Glob paths to ignore
+* @param {Boolean} [options.followSymlinks] follow symlinks
+* @returns {Function} pull stream source
+*/
+module.exports = (...args) => {
+  const options = isString(args[args.length - 1]) ? {} : args.pop()
```

Comment:
This should be enough we don't box strings 
```suggestion
  const options = typeof args[args.length - 1] === 'string' ? {} : args.pop()
```

Trying to avoid anything from lodash lots of unnecessary use of it that increase bundle size.